text/XXX-A-Compilation-of-Short-Interactive-Tutorials-and-How-To-Guides-for-Coq.md

```diff
@@ -0,0 +1,257 @@
+
+Yet, in our preliminary discussions, we have seen a need for this kind of content to ease the first steps in the ecosystem, in particular for introductory games like the [Natural Number Game](https://adam.math.hhu.de/#/g/leanprover-community/nng4) or the [Hott Game](https://homotopytypetheory.org/2021/12/01/the-hott-game/), that provide a nice introduction to Coq or to a concept and its very basic aspects in a fun and engaging way.
+
+Designing games is very different than designing tutorials or how-to guides, and will require a specific work on the subject.
+In particular, there may be more than one introductory game, depending on what is meant to be introduced, Coq and logic, HoTT, programming in Coq, and potentially on the background of the users.
+
+
+### 3.2 Horizontal Documentation
+Such a documentation has by nature the potential to be horizontal as most pairs of tutorials have no reason to be linked. Ensuring it by design is very important for the documentation to be accessible, and will allow by nature differentiated learning.
+
+#### Accessibility
+
+As we have already said, an effort should be made so that tutorials are as self-contained even if it means adding some reminders. Moreover, it is important that users do not have to know, e.g., how `lia` works to get to know how sections work and conversely. Similarly, it is important to be able to sandbox Coq's theory as much as possible. While it needs to be discussed as well, we do not wish for users to have to learn Coq's type theory to learn how to use sections or other such mechanisms. As reported by some Coq users we discussed with, when one asks something simple about Coq, many things can come all at once.
+
+#### Writing and Maintaining
+
+Contrarily to a very linear and structured course about Coq, horizontal tutorials and how-to guides make it easier for many authors to write in parallel documentation about very different aspects, and to add content progressively.
+
+#### Differentiated learning
+
+Being horizontal, such a documentation would enable to have different tutorials (for some specific features) for different users. For instance, one could have:
+
+- Different tutorials depending on the background, e.g. for mathematicians or engineers.
+- An Ltac2 tutorial for people that already know Ltac, and a tutorial for newcomers.
+- Different tutorials depending on the level of the user, e.g. a basic tutorial about inductive types, and a more advanced one discussing notions like strict positivity or the guard condition.
+
+It would also have the advantage to be (relatively) easy to maintain as the tutorials would mostly be independent, short, and in a reasonable number.
+
+
+### 3.3 Online interface
+
+The documentation would be interactive and available online, in a format more or less similar to [Pierre Rousselin's class](https://www.math.univ-paris13.fr/~rousselin/ipf/2023/RInegalit%C3%A9s_sujet.html).
```

Comment:
I don't see how the linked webpage is interactive.

Making interactive pages such as for the Natural Number Game above seems like a lot more work than just writing a non-interactive tutorial.  What are the relative priorities of interactive and non-interactive tutorials?

Reply:
You have to click on the JsCoq button on the right of your screen and enable JavaScript